compyle/array.py

```diff
@@ -456,11 +460,14 @@ def get_allocator(queue):
 
 
 def sort_by_keys(ary_list, out_list=None, key_bits=None,
-                 backend=None, use_radix_sort=False):
+                 backend=None, use_radix_sort=False,
+                 profile_id="sort_by_keys"):
     # FIXME: Need to use returned values, cuda backend uses
     # thrust that will internally allocate a new array for storing
     # the sorted data so out_list will not have the sorted arrays
     # first arg of ary_list is the key
+    result = None
+    start = time.time()
```

Comment:
This block looks like something perfect for a context manager.  So a `with profile('name'): ` may be a way to facilitate this?